src/utils/config_filters.cpp

```diff
@@ -25,6 +27,10 @@ bool utils::config_filters::bool_matches_if_present(const config& filter, const
 	if(!filter.has_attribute(attribute)) {
 		return true;
 	}
+	//check if filter is boolean, if not then show error message.
+	if(filter[attribute].to_bool(true) != filter[attribute].to_bool(false)) {
+		gui2::show_error_message(_("The filter $attribute can only take the values yes/true or no/false."));
```

Comment:
Why would displaying `$attribute` in the error message be useful?

Reply:
excuse me i would doing `show_error_message(_("The filter ") + attribute +  `

Reply:
@soliton Yes, I was wrong, no need to lower your thumb.

Reply:
That's also incorrect - for an example see https://github.com/wesnoth/wesnoth/blob/6b9a7627b90f90c0fc4f2065d19b179df4682280/src/chat_events.cpp#L57-L60

Reply:
itry but that don't work

Reply:
@Pentarctagon i pull that: `gui2::show_error_message(_("The filter ") + attribute +  _(" can only take the values yes/true or no/false, the check will therefore not be performed on this filter."));`

the method what you're suggest don't work.

Reply:
`gui2::show_error_message(_("The filter ") + attribute +  _(" can only take the values yes/true or no/false, the check will therefore not be performed on this filter."));` is not acceptable since that creates to separate translatable strings.